lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,455 @@
+        #        Values for limiting prob(lightning) with prob(precip).
+        #        These are the three prob(precip) thresholds and are designed
+        #        to prevent a large probability of lightning being output if
+        #        the probability of precipitation is very low.
+        self.precipthr = (0.0, 0.05, 0.1)
+        #    ltngthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip)
+        #        These are the three prob(lightning) values to scale to.
+        self.ltngthr = (0.0067, 0.25, 1.)
+        #    probability thresholds for increasing the prob(lightning)
+        #        phighthresh for heavy precip (>7mm/hr)
+        #            relates to problightning_values[2]
+        #        ptorrthresh for intense precip (>35mm/hr)
+        #            relates to problightning_values[1]
+        self.phighthresh = 0.4
+        self.ptorrthresh = 0.2
+        #    pl_dict (dict):
+        #        Lightning probability values to increase first-guess to if
+        #        the lightning_thresholds are exceeded in the nowcast data.
+        #        Dict must have keys 1 and 2 and contain float values.
+        #        The default values are selected to represent lightning risk
+        #        index values of 1 and 2 relating to the key.
+        self.pl_dict = {1: 1., 2: 0.25}
+
+        # Set values for handling vertically-integrated-ice (VII) data
+        #    ice_thresholds (tuple):
+        #        Values for increasing prob(lightning) with column-ice data.
+        #        These are the three VII thresholds in kg/m2.
+        self.ice_thresholds = (0.5, 1.0, 2.0)
+        #    ice_scaling (tuple):
+        #        Values for increasing prob(lightning) with VII data.
+        #        These are the three prob(lightning) values to scale to.
+        self.ice_scaling = (0.1, 0.5, 0.9)
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius}, debug={debug},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+>""".format(radius=self.radius, debug=self.debug,
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2])
+
+    def _process_haloes(self, cube):
+        """
+        Adjust data so that lightning probability does not decrease too rapidly
+        with distance.
+
+        Args:
+            cube (iris.cube.Cube):
+                Radius will be applied equally on x and y dimensions.
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube of same shape as cube with haloes applied.
+        """
+        new_cube = self.neighbourhood.process(cube.copy())
+        return new_cube
```

Comment:
Still wondering why we need this wrapper.  The information from the docstring could be included as an inline comment where the neighbourhood processing is called from `process`.  This would reduce complexity.